core/storage/trie/polkadot_trie_db/polkadot_trie.hpp

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_POLKADOT_TRIE_HPP
+#define KAGOME_POLKADOT_TRIE_HPP
+
+#include "storage/face/generic_map.hpp"
+#include "storage/trie/polkadot_trie_db/polkadot_codec.hpp"
+#include "storage/trie/polkadot_trie_db/polkadot_node.hpp"
+
+namespace kagome::storage::trie {
+
+  /**
+   * For specification see
+   * https://github.com/w3f/polkadot-re-spec/blob/master/polkadot_re_spec.pdf
+   * 5.2 The General Tree Structure and further
+   */
+  class PolkadotTrie
+      : public face::ReadableMap<common::Buffer, common::Buffer>,
+        public face::WriteableMap<common::Buffer, common::Buffer> {
+    using NodePtr = std::shared_ptr<PolkadotNode>;
+    using BranchPtr = std::shared_ptr<BranchNode>;
+    using ChildRetrieveCallback =
+        std::function<outcome::result<NodePtr>(BranchPtr, uint8_t)>;
+
+    inline static outcome::result<NodePtr> DefaultChildRetrieveCallback(
```

Comment:
Can't you move the definition to .cpp? Also, comment it. Also, name it from the lowcase

Reply:
I commented it and renamed it, but I think that its definition may stay here. It doesn't spoil the class declaration as it is justo one short line long, but it serves as an evident example of what child retrieve callback is.